spec/models/user_spec.rb

```diff
@@ -1,5 +1,45 @@
 require 'rails_helper'
 
 RSpec.describe User, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  user = User.new(name: 'Peter',
+                  photo: 'photo_url',
+                  bio: 'This is a bio for Mr. Peter!',
+                  posts_counter: 0)
+
+  before { user.save }
+
+  describe 'validate data: ' do
+    it 'name should be present' do
+      user.name = ''
+      expect(user).to_not be_valid
+    end
+
+    it 'posts_counter should be greater than or equal to zero' do
+      user.name = 'John'
+      user.posts_counter = -1
+      expect(user).to_not be_valid
+    end
+
+    it 'posts_counter value should be positive integer' do
+      user.name = 'John'
+      user.posts_counter = 'one'
+      expect(user).to_not be_valid
+    end
+  end
+
+  describe 'most_recent_posts method' do
+    user1 = User.first
+
+    before do
+      user1.posts.create(title: 'post-title1', text: 'post body one')
+      user1.posts.create(title: 'post-title2', text: 'post body two')
+      user1.posts.create(title: 'post-title3', text: 'post body three')
+      user1.posts.create(title: 'post-title4', text: 'post body four')
+    end
+
+    it 'should return up to three latest posts' do
+      expect(user1.most_recent_posts.length).to be <= 3
+      expect(user1.most_recent_posts[0].title).to eq 'post-title4'
+    end
+  end
```

Comment:
- [ ] These last two tests failed `undefined method 'posts' for NilClass`, which means that the `user1` does not exist. You can fix this by making sure that `User` is successfully created before each test case. You can use `subject`, which returns a created user for your tests. See what your test looks like when you use it below, But if you have another way of making the last two tests pass, that's okay.

- [ ] Also, please remove the less than sign `<` you have on line `41` when testing for `most_recent_posts.length`. It should be equal to `3` since you have more than three post created. This will make the test more accurate.

```ruby
  RSpec.describe User, type: :model do
	subject do
		User.new(name: 'Peter', photo: 'photo_url', bio: 'This is a bio for Mr. Peter!', posts_counter: 0)
	end

  before { subject.save }

  describe 'validate data: ' do
    it 'name should be present' do
      subject.name = ''
      expect(subject).to_not be_valid
    end

    it 'posts_counter should be greater than or equal to zero' do
      subject.name = 'John'
      subject.posts_counter = -1
      expect(subject).to_not be_valid
    end

    it 'posts_counter value should be positive integer' do
      subject.name = 'John'
      subject.posts_counter = 'one'
      expect(subject).to_not be_valid
    end
  end

  describe 'most_recent_posts method' do
    before do
      subject.posts.create(title: 'post-title1', text: 'post body one')
      subject.posts.create(title: 'post-title2', text: 'post body two')
      subject.posts.create(title: 'post-title3', text: 'post body three')
      subject.posts.create(title: 'post-title4', text: 'post body four')
    end

    it 'should return up to three latest posts' do
      expect(subject.most_recent_posts.length).to eq(3)
      expect(subject.most_recent_posts[0].title).to eq 'post-title4'
    end
  end
end
```